massa-bootstrap/src/server/white_black_list.rs

```diff
@@ -0,0 +1,114 @@
+use std::{
+    borrow::Cow,
+    collections::HashSet,
+    net::{IpAddr, SocketAddr},
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use massa_logging::massa_trace;
+use parking_lot::RwLock;
+
+use crate::tools::normalize_ip;
+
+/// A wrapper around the white/black lists that allows efficient sharing between threads
+// TODO: don't clone the path-bufs...
+#[derive(Clone)]
+pub(crate) struct SharedWhiteBlackList<'a> {
+    inner: Arc<RwLock<WhiteBlackListInner>>,
+    white_path: Cow<'a, Path>,
+    black_path: Cow<'a, Path>,
+}
+
+impl SharedWhiteBlackList<'_> {
+    pub(crate) fn new(white_path: PathBuf, black_path: PathBuf) -> Result<Self, String> {
+        let (white_list, black_list) =
+            WhiteBlackListInner::load_white_black_lists(&white_path, &black_path)?;
+        Ok(Self {
+            inner: Arc::new(RwLock::new(WhiteBlackListInner {
+                white_list,
+                black_list,
+            })),
+            white_path: Cow::from(white_path),
+            black_path: Cow::from(black_path),
+        })
+    }
+
+    /// Checks if the white/black list is up to date with a read-lock
+    /// Creates a new list, and replaces the old one in a write-lock
+    pub(crate) fn update(&mut self) -> Result<(), String> {
+        let read_lock = self.inner.read();
+        let (new_white, new_black) =
+            WhiteBlackListInner::load_white_black_lists(&self.white_path, &self.black_path)?;
+        let white_delta = new_white != read_lock.white_list;
+        let black_delta = new_black != read_lock.black_list;
```

Comment:
```rust 
let should_update = new_black.ne(&read_lock.black_list);
``` 
doesn"t work correctly ?

Reply:
I prefer to use `!=` as I find it less cognitive load for reading. using the method means that, at first glance, it could be _any_ method, then your brain has to process that it's a comparison operation, and it can take a non-zero cognitive load to process which is "left" and which is "right". the `!=` does away with a lot of that effort.